magento/magento2
# partially backported MAGETWO-83560 : Fixes issue with generating 301 rewrite when changing product URL key on Storeview level

### Description
Backport of MAGETWO-83560 for 2.1.x projects.
Fixes issue with generating 301 redirect when changing product URL key on Storeview level

### Fixed Issues (if relevant)

1. magento/magento2#10073: Magento don't create product redirect if URL key on store view level was changed.
2. magento/magento2#13240: Permanent 301 redirect is not generated when product url changes on storeview scope

### Manual testing scenarios
1. ... Change Product url key on storeview level
2. ... Check Url Rewrites table for the correct rewrites

### Contribution checklist
